src/validation/validation.cpp

```diff
@@ -2949,7 +2934,11 @@ bool DisconnectTip(CValidationState &state, const Consensus::Params &consensusPa
     // place all tx back into the admission queue. "Rollbackchain" is used for significant manually triggered
     // reorganizations, such as switching between forks, so it makes no sense to keep the transactions because
     // they will likely be invalid or already confirmed on the other fork.
-    if (fRollBack)
+    // Do the same If this block enabled the Nov 15th 2019 protocol upgrade, then we need to clear the
+    // mempool and transactions commit queue of any transaction using not previously avaiable features
+    // (e.g. Schnorr enabled OP_CHECKMULTISIG)
+    if (fRollBack ||
+        (IsNov2019Enabled(consensusParams, pindexDelete) && !IsNov2019Enabled(consensusParams, pindexDelete->pprev)))
     {
```

Comment:
You don't need to add this check for the fork here.  It doesn't matter anymore whether we are forking or not the mempool gets cleared and resubmitted in the else statement.

Reply:
I add the check here because I wanted to be sure that we won't revalidate txns which were submitted and validated according to a different set of consensus rules. 

In hindsight I should have used just `IsNov2019Enabled(consensusParams, pindexDelete)` so that we discard all txns committed to the chain since fork activation.

Do you think it's worth re-submit for validation txns that have been validated according to a different set of consensus rules?

Reply:
I think we should resubmit them because most of them will still be valid and very few of them would likely get discarded because of different consensus rules.

Reply:
ok will do.